cinder/tests/unit/attachments/test_attachments_manager.py

```diff
@@ -243,12 +243,13 @@ def test_connection_terminate_no_connector_force_false(self):
         # does not have a connector will not call the driver and return None
         # if the force flag is False.
         attachment = mock.MagicMock(connector={})
+        volume = mock.MagicMock()
         with mock.patch.object(self.manager.driver, '_initialized',
                                create=True, new=True):
             with mock.patch.object(self.manager.driver,
                                    'terminate_connection') as term_conn:
                 has_shared_connection = self.manager._connection_terminate(
-                    self.context, mock.sentinel.volume, attachment)
+                    self.context, volume, attachment)
```

Comment:
Will you do this @hemna or won't you.

Reply:
because sentinel objects don't have a get attribute, which is what is used inside the oslo log library for getting attributes out of the args, kwargs passed in to log. MagicMock covers attributes and functions as we are not testing the internals of oslo log here.

```
File "/home/waboring/.pyenv/versions/3.8.10/lib/python3.8/logging/__init__.py", line 1806, in info
    self.log(INFO, msg, *args, **kwargs)

      File "/home/waboring/.pyenv/versions/3.8.10/lib/python3.8/logging/__init__.py", line 1843, in log
    msg, kwargs = self.process(msg, kwargs)

      File "/home/waboring/cinder/.tox/py38/lib/python3.8/site-packages/oslo_log/log.py", line 176, in process
    if not resource.get('name', None):

    AttributeError: '_SentinelObject' object has no attribute 'get'
 ```